### ISSUE TITLE
aws: public-ipv4 is not always present, especially in VPC environments

### ISSUE THREAD

**Issue by c4milo** [hashicorp/nomad#611]
When running nomad node-status <my node ID> I get the following: 

```
[root@per-nomad-server01 ~]# nomad node-status bb7cea73-0638-e4e9-4940-cf795e74a0fc
ID         = bb7cea73-0638-e4e9-4940-cf795e74a0fc
Name       = per-nomad-worker03
Class      = <none>
Datacenter = iad2
Drain      = false
Status     = ready
Attributes = arch:amd64, consul.datacenter:iad2, consul.name:per-nomad-worker03, consul.revision:46499d6e7237ca8547c15ce44e9b93bea48a455b, consul.server:false, consul.version:0.6.0, cpu.frequency:2500.024000, cpu.modelname:Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz, cpu.numcores:1, cpu.totalcompute:2500.024000, driver.docker:1, driver.docker.version:1.9.1, driver.exec:1, hostname:per-nomad-worker03, kernel.name:linux, kernel.version:3.10.0-229.14.1.el7.x86_64, memory.totalbytes:1928470528, network.ip-address:10.213.45.88, os.name:centos, os.version:7.2.1511, platform.aws.ami-id:ami-83fbbee6, platform.aws.hostname:ip-10-213-45-88.iad2.foo.net, platform.aws.instance-id:i-a1442017, platform.aws.instance-type:t2.small, platform.aws.local-hostname:ip-10-213-45-88.iad2.foo.net, platform.aws.local-ipv4:10.213.45.88, platform.aws.placement.availability-zone:us-east-1b, platform.aws.public-hostname:<?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
         "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
  <title>404 - Not Found</title>
 </head>
 <body>
  <h1>404 - Not Found</h1>
 </body>
</html>, platform.aws.public-ipv4:<?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
         "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
  <title>404 - Not Found</title>
 </head>
 <body>
  <h1>404 - Not Found</h1>
 </body>
</html>, storage.bytesfree:6870028288, storage.bytestotal:8578400256, storage.volume:/dev/xvda1
```

**Comment by c4milo** [comment]
I guess the issue is about handling 404 errors when requesting AWS metadata.

**Comment by dadgar** [comment]
Should be fixed by #614
